File: src/simple_model_test.zig
// simple_model_test.zig - Simple test for model embedding functionality
const std = @import("std");
const model_embedding = @import("model_embedding.zig");

pub fn main() !void {
    std.debug.print("=== Simple Model Embedding Test ===\n", .{});
    
    // Test basic model access
    const model_data = model_embedding.getEmbeddedModel();
    const model_size = model_embedding.getModelSize();
    
    std.debug.print("Model data pointer: {any}\n", .{model_data.ptr});
    std.debug.print("Model size: {} bytes\n", .{model_size});
    
    // Test model validation
    const is_valid_gguf = model_embedding.ModelValidator.validateGGUF(model_data);
    const is_valid_header = model_embedding.ModelValidator.validateHeader(model_data);
    
    std.debug.print("GGUF format validation: {}\n", .{is_valid_gguf});
    std.debug.print("Header validation: {}\n", .{is_valid_header});
    
    // Test model loader
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    defer _ = gpa.deinit();
    const allocator = gpa.allocator();
    
    var loader = model_embedding.ModelLoader.init(allocator);
    const loaded_model = try loader.loadEmbeddedModel();
    
    std.debug.print("Loaded model size: {} bytes\n", .{loaded_model.len});
    
    const is_loaded_valid = try loader.validateModel(loaded_model);
    std.debug.print("Loaded model validation: {}\n", .{is_loaded_valid});
    
    // Test model metadata
    const metadata = model_embedding.getModelMetadata();
    
    std.debug.print("Magic number: {s}\n", .{metadata.magic});
    std.debug.print("Version: {}\n", .{metadata.version});
    std.debug.print("Tensor count: {}\n", .{metadata.tensor_count});
    std.debug.print("KV count: {}\n", .{metadata.kv_count});
    std.debug.print("Architecture: {s}\n", .{metadata.architecture});
    std.debug.print("Context length: {}\n", .{metadata.context_length});
    
    std.debug.print("\n=== Test completed successfully! ===\n", .{});
}

File: src/simple_model_test.c
// simple_model_test.c - Simple test for model embedding
#include <stdio.h>
#include <stdlib.h>
#include "../include/c_model_interface.h"

int main() {
    printf("=== Simple Model Embedding Test ===\n");
    
    // Test getting embedded model info
    printf("1. Testing embedded model access...\n");
    const uint8_t* model_data = c_get_embedded_model();
    size_t model_size = c_get_embedded_model_size();
    
    printf("   Model data pointer: %p\n", (void*)model_data);
    printf("   Model size: %zu bytes\n", model_size);
    
    // Test model validation
    printf("2. Testing model validation...\n");
    int is_valid = c_validate_embedded_model();
    printf("   Model validation result: %s\n", is_valid ? "VALID" : "INVALID");
    
    // Test loading model info
    printf("3. Testing model loading...\n");
    embedded_model_info_t model_info = c_load_embedded_model();
    printf("   Loaded model info:\n");
    printf("     Data pointer: %p\n", (void*)model_info.data);
    printf("     Size: %zu bytes\n", model_info.size);
    printf("     Valid: %s\n", model_info.is_valid ? "YES" : "NO");
    
    // Test GGUF format validation
    printf("4. Testing GGUF format validation...\n");
    int format_valid = c_validate_gguf_format(model_info.data, model_info.size);
    printf("   GGUF format validation: %s\n", format_valid ? "VALID" : "INVALID");
    
    printf("\n=== Test completed successfully! ===\n");
    return 0;
}